inference-engine/thirdparty/clDNN/src/loop.cpp

```diff
@@ -189,6 +189,104 @@ static void validate_mappings(loop_node const & node) {
+    if (!preproc_memories_done) {
+        return;
+    }
+    // update output memory
+    const auto& output_primitive_maps = node.get_output_primitive_maps();
+    for (size_t i = 0; i < output_primitive_maps.size(); ++i) {
+        const auto& output_mapping = output_primitive_maps.at(i);
+        const primitive_id& external_id = output_mapping.external_id;
+        const primitive_id& internal_id = output_mapping.internal_id;
+        memory::ptr to_mem = get_external_memory(external_id);
+        if (output_mapping.axis < 0) {
+            body_network->get_primitive(internal_id)->set_output_memory(to_mem);
+        } else {
+            for (auto& mem_mapping : concatenated_output_mem_mappings) {
+                if (mem_mapping.concat_data_prim->id() == internal_id) {
+                    mem_mapping.concatenated_mem = to_mem;
+                    break;
+                }
+            }
+        }
+    }
+    // update input memory
+    for (size_t memory_num = 0; memory_num < inputs_memory_count(); memory_num++) {
+        const primitive_id& input_external_id = dependencies().at(memory_num)->id();
+        auto input_map_ptrs = node.find_io_primitive_maps(input_external_id, true);
+        if (input_map_ptrs.size() == 0) {
+            if (input_external_id == node.get_trip_count_id() ||
+                input_external_id == node.get_initial_execution_id()) {
+                continue;
+            }
+        }
+
+        auto memory = input_memory_ptr(memory_num);
+        for (size_t i = 0; i < input_map_ptrs.size(); ++i) {
+            const auto input_map = input_map_ptrs.at(i);
+            bool is_concatenated_input = (input_map->axis >= 0);
+            if (is_concatenated_input) {
+                for (auto& mem_mapping : concatenated_input_mem_mappings) {
+                    if (mem_mapping.sliced_data_prim->id() == input_map->internal_id) {
+                        mem_mapping.concatenated_mem = memory;
+                        break;
+                    }
+                }
+            } else {
+                body_network->set_input_data(input_map->internal_id, memory);
+            }
+        }
+    }
+    //update backedges memory
+    const auto& back_edges = node.get_back_edges();
+    // checking if memory is a destination of a backedge
+    for (const auto& back_edge : back_edges) {
+        //find corresponding input of the backedge
+        const auto input_map_ptrs = node.find_io_primitive_maps(back_edge.to, false);
+        assert(input_map_ptrs.size() == 1);
+        const auto& input_map = input_map_ptrs.front();
+        auto backedged_sliced_output_mems = get_sliced_mem(back_edge.from);
+        const auto backedge_to_prim = body_network->get_primitive(back_edge.to);
+        const auto backedge_from_prim = body_network->get_primitive(back_edge.from);
+        memory::ptr initial_mem = get_external_memory(input_map->external_id);
+
+        for (auto& backedge_mapping : backedge_memory_mappings) {
```

Comment:
seems that it would be better to have a new function to get the corresponding memory mapping, but not a big problem, so it is okay for now I think.